### PR TITLE
[Google Blockly] Add High Contrast Theme

### PR DESCRIPTION
**Pull request by mikeharv**
For more context:
* https://github.com/code-dot-org/code-dot-org/pull/50243
* https://github.com/code-dot-org/code-dot-org/pull/50246
* https://github.com/code-dot-org/code-dot-org/pull/50264

Similar to the work to add a "dark" theme to the context menu (above PRs), we are now ready to add a "high contrast" theme. Blockly provides a plugin that works as a starting place: 
https://github.com/google/blockly-samples/tree/master/plugins/theme-highcontrast
The plugin includes customizations font and component styles. We simply need to override the block styles.

After meeting with the Design team, we agreed to use the [Adobe Color contrast checker analyzer tool | Adobe Color](https://color.adobe.com/create/color-contrast-analyzer) to select replacement colors. These colors are collected here: https://docs.google.com/spreadsheets/d/1VAasuYOJIdUP3A-hZBsxK5dcf8CAlFLNsMIDxZtRlNI/edit#gid=0

image
image

## Links

Google Sheet: [Color Palettes for Accessible Blockly Themes](https://docs.google.com/spreadsheets/d/1VAasuYOJIdUP3A-hZBsxK5dcf8CAlFLNsMIDxZtRlNI/edit#gid=0)

jira ticket: [SL-441: [Google Blockly] Add themes for accessibility](https://codedotorg.atlassian.net/browse/SL-441)

Accessibility Tool: [Adobe Color contrast checker analyzer tool | Adobe Color](https://color.adobe.com/create/color-contrast-analyzer)

## Follow-up work

Add more themes for color vision deficiency.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/blockly/addons/contextMenu.js [hunk 1]
```diff
@@ -219,6 +220,38 @@ const registerDarkTheme = function() {
...
+      } else {
+        return ENABLED;
+      }
+    },
+    callback: function(scope) {
+      localStorage.setItem(BLOCKLY_THEME, HIGH_CONTRAST);
+      scope.workspace.setTheme(Blockly.themes.highContrast);
+    },
+    scopeType: GoogleBlockly.ContextMenuRegistry.ScopeType.WORKSPACE,
+    id: 'highContrastTheme',
```
Comment (sanchitmalhotra126) [inline apps/src/blockly/addons/contextMenu.js:246]: Is this theme ID referenced anywhere else? Should we use a constant for it?

File: apps/src/blockly/themes/cdoHighContrast.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  ...danceHighContrastStyles,
+  ...musicLabHighContrastStyles
+};
+
+export const cdoHighContrastBlockStyles = {
+  ...coreBlocklyOverrides,
+  ...cdoCustomHighContrastStyles
+};
+
+export default GoogleBlockly.Theme.defineTheme('cdohighcontrast', {
```
Comment (sanchitmalhotra126) [inline apps/src/blockly/themes/cdoHighContrast.js:75]: Same here, for `'cdohighcontrast'`?

**Review by sanchitmalhotra126 — APPROVED** [review]
Just a few questions on constants, otherwise looks good!

File: apps/src/blockly/addons/contextMenu.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ const HIDDEN = 'hidden';
 const MODERN = 'modern';
 const DARK = 'dark';
 const MUSICLAB_DARK = 'musiclabdark';
+const HIGH_CONTRAST = 'highContrast';
```
Comment (fisher-alice) [inline apps/src/blockly/addons/contextMenu.js]: What do you think of storing these constants in `blockly/constants.js`? You could do something like this...
```
export const ThemeOptions = {
  ENABLED: 'enabled',
  DISABLED: 'disabled',
  HIDDEN: 'hidden'
};
```
Could do something similar for the themes as well.

**Review by fisher-alice — APPROVED** [review]
I checked out your branch and it looks great. Using the context menu was a really smooth experience. 🎉 
Just added a comment about the constants. Nice job!

File: apps/src/blockly/addons/contextMenu.js (hunk 1 above)
Comment (mikeharv) [reply apps/src/blockly/addons/contextMenu.js:246]: It's not. This specific id is just for the context menu option element.

File: apps/src/blockly/addons/contextMenu.js (hunk 3 above)
Comment (mikeharv) [reply apps/src/blockly/addons/contextMenu.js]: Thanks Alice! I took your suggestion and made one tweak to names to better reflect what enabled/hidden/disabled are used for. I was also able to work the theme constants into the wrapper: https://github.com/code-dot-org/code-dot-org/pull/50297/commits/3bb1990b69a30a834863df33d9adbed781f47bd5

Comment (fisher-alice) [reply apps/src/blockly/addons/contextMenu.js]: Awesome! Thanks for the update!

**Review by fisher-alice — APPROVED** [review]
Thanks for the updates!

**Pull request merged**